src/edtr-io/plugins/paste-hack/editor.tsx

```diff
@@ -0,0 +1,84 @@
+
+export const PasteHackEditor: React.FunctionComponent<PasteHackPluginProps> = (
+  props
+) => {
+  const store = useScopedStore()
+
+  function replaceWithStateString() {
+    console.log('starting replace')
+
+    // TODO: Use io-ts here
+    const content = JSON.parse(dummyContent) as {
+      plugin: 'rows'
+      state: Document[]
+    }
```

Comment:
Which types do we want to accept as input?! Examples:

* `type State = { plugin: string, state: unknown }`
* `{ plugin: "rows", state: State[] }`
* `State[]`

Reply:
I tried to make the type more explicit in the io-ts decoder. Basically `{ plugin: "rows", state: State[] }` with a check if the rows child plugins are supported.